code/modules/mob/living/simple_animal/revenant/revenant.dm

```diff
@@ -196,21 +266,33 @@
 
 
 
-/mob/living/simple_animal/revenant/proc/reveal(time, stun)
+/mob/living/simple_animal/revenant/proc/reveal(time)
 	var/mob/living/simple_animal/revenant/R = usr
 	if(!istype(usr) || !usr)
 		return
+	if(time <= 0)
+		return
 	R.revealed = 1
 	R.invisibility = 0
-	if(stun)
-		R.notransform = 1
-	R << "<span class='warning'>You have been revealed[stun ? " and cannot move" : ""].</span>"
+	R << "<span class='warning'>You have been revealed.</span>"
 	spawn(time)
 		R.revealed = 0
 		R.invisibility = INVISIBILITY_OBSERVER
-		if(stun)
-			R.notransform = 0
-		R << "<span class='notice'>You are once more concealed[stun ? " and can move again" : ""].</span>"
+		R << "<span class='notice'>You are once more concealed.</span>"
+
+/mob/living/simple_animal/revenant/proc/stun(time)
+	var/mob/living/simple_animal/revenant/R = usr
+	if(!istype(usr) || !usr)
```

Comment:
```!usr``` is unnecessary, if ```usr``` is ```null``` it will also fail the ```istype()```, meaning the ```if(!usr)``` part is never called.